control/pure_pursuit/src/pure_pursuit/pure_pursuit_lateral_controller.cpp

```diff
@@ -365,24 +340,46 @@ boost::optional<Trajectory> PurePursuitLateralController::generatePredictedTraje
   return predicted_trajectory;
 }
 
-boost::optional<LateralOutput> PurePursuitLateralController::run()
+bool PurePursuitLateralController::isReady(const InputData & input_data)
 {
   current_pose_ = self_pose_listener_.getCurrentPose();
+  trajectory_ = input_data.current_trajectory;
+  current_odometry_ = input_data.current_odometry;
+  current_steering_ = input_data.current_steering;
+
   if (!isDataReady()) {
-    return boost::none;
+    return false;
   }
   setResampledTrajectory();
   if (!output_tp_array_ || !trajectory_resampled_) {
-    return boost::none;
+    return false;
   }
   if (param_.enable_path_smoothing) {
     averageFilterTrajectory(*trajectory_resampled_);
   }
   const auto cmd_msg = generateOutputControlCmd();
   if (!cmd_msg) {
     RCLCPP_ERROR(node_->get_logger(), "Failed to generate control command.");
-    return boost::none;
+    return false;
   }
+
+  return true;
+}
+
+LateralOutput PurePursuitLateralController::run(const InputData & input_data)
+{
+  // TODO(murooka) needs to be refactored to seperate isReady and run clearly
+  current_pose_ = self_pose_listener_.getCurrentPose();
+  trajectory_ = input_data.current_trajectory;
+  current_odometry_ = input_data.current_odometry;
+  current_steering_ = input_data.current_steering;
```

Comment:
If the input date is set in the member variable in several functions, it is more reasonable to have `setData()` function to me. I don't expect `isReady()` function stores the input data.

Reply:
I agree. To fix this, mpc and pure_pursuit should be refactored which is the reason I skipped it in this PR.

FYI: In pid_longitudinal_controller, the variable is set in `run` function
![image](https://user-images.githubusercontent.com/20228327/208821055-4863211d-1cba-48fb-8524-74e4203f50e5.png)